x-pack/plugins/ml/public/application/components/delete_job_check_modal/delete_job_check_modal.tsx

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useState, useEffect } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiModal,
+  EuiOverlayMask,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiButton,
+  EuiLoadingSpinner,
+} from '@elastic/eui';
+import { JobType, CanDeleteJobResponse } from '../../../../common/types/saved_objects';
+import { useMlApiContext } from '../../contexts/kibana';
+import { useToastNotificationService } from '../../services/toast_notification_service';
+
+const shouldUnTagLabel = i18n.translate('xpack.ml.deleteJobCheckModal.shouldUnTagLabel', {
+  defaultMessage: 'Remove job from current space',
+});
+
+interface ModalContentReturnType {
+  buttonText: JSX.Element;
+  modalText: JSX.Element;
+}
+
+interface JobCheckRespSummary {
+  canDelete: boolean;
+  canUntag: boolean;
+  canTakeAnyAction: boolean;
+}
+
+function getRespSummary(resp: CanDeleteJobResponse): JobCheckRespSummary {
+  const jobsChecked = Object.keys(resp);
+  // Default to first job's permissions
+  const { canDelete, canUntag } = resp[jobsChecked[0]];
+  let canTakeAnyAction = true;
+
+  if (jobsChecked.length > 1) {
+    // Check all jobs and make sure they have the same permissions - otherwise no action can be taken
+    canTakeAnyAction = jobsChecked.every(
+      (id) => resp[id].canDelete === canDelete && resp[id].canUntag === canUntag
+    );
+  }
+
+  return { canDelete, canUntag, canTakeAnyAction };
+}
+
+function getModalContent(
+  jobIds: string[],
+  respSummary: JobCheckRespSummary
+): ModalContentReturnType {
+  const { canDelete, canUntag, canTakeAnyAction } = respSummary;
+
+  if (canTakeAnyAction === false) {
+    return {
+      buttonText: (
+        <FormattedMessage
+          id="xpack.ml.deleteJobCheckModal.buttonTextNoAction"
+          defaultMessage="Close"
+        />
+      ),
+      modalText: (
+        <FormattedMessage
+          id="xpack.ml.deleteJobCheckModal.modalTextNoAction"
+          defaultMessage="{ids} have different space permissions. When you delete multiple jobs, they must have the same permissions. De-select those jobs then re-try the deletion."
+          values={{ ids: jobIds.join(', ') }}
+        />
+      ),
+    };
+  }
+
+  const noActionContent: ModalContentReturnType = {
+    buttonText: (
+      <FormattedMessage id="xpack.ml.deleteJobCheckModal.buttonTextClose" defaultMessage="Close" />
+    ),
+    modalText: (
+      <FormattedMessage
+        id="xpack.ml.deleteJobCheckModal.modalTextClose"
+        defaultMessage="{ids} cannot be deleted and cannot be removed from the current space."
```

Comment:
First time I saw this message I was unclear on the reason for it. I think it would be worth adding an extra bit of text with the reason, such as `because this job is assigned to the * space and you do not have access to all spaces`.